external/source/exploits/CVE-2019-0808/NtUserMNDragOverExploit/dllmain.cpp

```diff
@@ -0,0 +1,668 @@
+		if (payload[0] != 'P' && payload[1] != 'A') {
+			LPVOID shellcode = VirtualAlloc(NULL, sizeof(payload), MEM_COMMIT | MEM_RESERVE, PAGE_EXECUTE_READWRITE);
+			memcpy(shellcode, payload, sizeof(payload));

```

Comment:
Where is the check here to ensure `VirtualAlloc()` didn't end up failing before we call `memcpy()`?

Reply:
Confirmed fixed, `shellcode` is now checked to ensure it is not NULL before continuing. This works as `if (NULL)` is the same as `if (FALSE)` in C++.